### PR TITLE
Update gen_fbin_file.cpp to support binary data type

### PR DESCRIPTION
**Pull request by cydrain**

### CODE REVIEW

File: benchmark/hdf5/gen_fbin_file.cpp [hunk 1]
```diff
@@ -44,20 +44,30 @@ class Create_FBIN : public Benchmark_hdf5, public ::testing::Test {
     TearDown() override {
     }
 
+    template <bool is_binary>
```
Comment (alexanderguzhva) [inline benchmark/hdf5/gen_fbin_file.cpp:47]: This overcomplicates the code.
Please put the new functionality into new functions, such as `fbin_write_binary()` or `fbin_read_binary()`.

**Review by alexanderguzhva — CHANGES_REQUESTED** [review]

File: benchmark/hdf5/gen_fbin_file.cpp (hunk 1 above)
Comment (cydrain) [reply benchmark/hdf5/gen_fbin_file.cpp]: updated

**Pull request merged**